### ISSUE TITLE
Questions about training for multiple shapes and inference

### ISSUE THREAD

**Issue by shylee2021** [nv-tlabs/nglod#19]
Hello, thanks for the wonderful works and code!
While trying to reproduce this work for shapenet150, I've got several questions.

1) It seems that the released code is for training network for single mesh data. How should the training for multiple shapes be done? AFAI understand, each shape has its own sparse feature (defined with `FeatureVolume` in the code). Am I getting right so that I have to train individual shape with shared mlp and different feature volume, or the model is the same through the entire dataset, which means feature volume is also shared?

1-1) If feature volume is also trained for all data, how is the minibatch consist? Should the shape be sampled from dataset so that dimension is `(num_shape) x (num_samples) x (dimension)`? Or should the point be sampled from dataset regardless of the shape? (in this case, the minibatch dim will be `(num_samples) x (dimension)`)

2) What is the inference process when we have fully trained model and arbitrary pointcloud which does not have groundtruth sdf value? In case of DeepSDF, the trained net is fixed and feature vector is trained with slightly different loss function when inferring.

Sorry for the broken english. If you don't understand those questions, I'll explain in detail.
Thanks in advance!

**Comment by tovacinni** [comment]
Thanks for your interest in our work!

1. If you want to use a sparse features, you'd probably have to have a list of feature volumes where each correspond to a different shape. The MLP can be shared across different networks, but the feature volume can't be shared.

1-1. We don't directly support multi-shape training in the code we release, but both shapes you describe shoudl work. DeepSDF for example I believe does the former and shows that it works okay.

2. For an arbitrary pointcloud, you'd probably have to estimate the SDF on non-surface points. We also don't directly support this but you should be able to do this with some smart losses, like Eikonal regularization.

Let us know if you have any other questions!

**Comment by shylee2021** [comment]
Thank you for answering!

**Comment by rmraaron** [comment]
> 

Hi, 

Just to ask for a quick update, did you train this network on multiple shapes? Thanks!

**Comment by joeylitalien** [comment]
Hi @rmraaron,

The decoder networks can generalized to multiple shapes (as shown in the paper), but the feature grid needs to be optimized per shape.

**Comment by rmraaron** [comment]
@joeylitalien Thanks for your quick reply.
